toolchain/semantics/semantics_context.cpp

```diff
@@ -263,6 +263,186 @@ auto Context::is_current_position_reachable() -> bool {
   }
 }
 
+auto Context::Initialize(ParseTree::Node parse_node, SemIR::NodeId target_id,
+                         SemIR::NodeId value_id) -> void {
+  // Implicitly convert the value to the type of the target.
+  auto type_id = semantics_ir().GetNode(target_id).type_id();
+  auto expr_id = ImplicitAsRequired(parse_node, value_id, type_id);
+  SemIR::Node expr = semantics_ir().GetNode(expr_id);
+
+  // Perform initialization now that we have an expression of the right type.
+  switch (SemIR::GetExpressionCategory(semantics_ir(), expr_id)) {
+    case SemIR::ExpressionCategory::NotExpression:
+      CARBON_FATAL() << "Converting non-expression node " << expr
+                     << " to initializing expression";
+
+    case SemIR::ExpressionCategory::DurableReference:
+    case SemIR::ExpressionCategory::EphemeralReference:
+      // The design uses a custom "copy initialization" process here. We model
+      // that as value binding followed by direct initialization.
+      //
+      // TODO: Determine whether this is observably different from the design,
+      // and change either the toolchain or the design so they match.
+      expr_id = AddNode(SemIR::Node::BindValue::Make(expr.parse_node(),
+                                                     expr.type_id(), expr_id));
+      [[fallthrough]];
+
+    case SemIR::ExpressionCategory::Value:
+      // TODO: For class types, use an interface to determine how to perform
+      // this operation.
+      AddNode(SemIR::Node::Assign::Make(expr.parse_node(), target_id, expr_id));
+      return;
+
+    case SemIR::ExpressionCategory::Initializing:
+      MarkInitializerFor(expr_id, target_id);
+      return;
+  }
+}
+
+auto Context::ConvertToValueExpression(SemIR::NodeId expr_id) -> SemIR::NodeId {
+  switch (SemIR::GetExpressionCategory(semantics_ir(), expr_id)) {
+    case SemIR::ExpressionCategory::NotExpression:
+      CARBON_FATAL() << "Converting non-expression node "
+                     << semantics_ir().GetNode(expr_id)
+                     << " to value expression";
+
+    case SemIR::ExpressionCategory::Initializing: {
+      // Commit to using a temporary for this initializing expression.
+      // TODO: Don't create a temporary if the initializing representation is
+      // already a value representation.
+      expr_id = FinalizeTemporary(expr_id, /*discarded=*/false);
+      [[fallthrough]];
+    }
```

Comment:
I think braces aren't needed here?

```suggestion
    case SemIR::ExpressionCategory::Initializing:
      // Commit to using a temporary for this initializing expression.
      // TODO: Don't create a temporary if the initializing representation is
      // already a value representation.
      expr_id = FinalizeTemporary(expr_id, /*discarded=*/false);
      [[fallthrough]];
```